backend/src/schema/resolvers/follow.spec.js

```diff
@@ -1,36 +1,93 @@
-import { GraphQLClient } from 'graphql-request'
+import { createTestClient } from 'apollo-server-testing'
 import Factory from '../../seed/factories'
-import { host, login } from '../../jest/helpers'
+import { getDriver } from '../../bootstrap/neo4j'
+import createServer from '../../server'
+import { gql } from '../../jest/helpers'
 
 const factory = Factory()
-let clientUser1
-let headersUser1
+const driver = getDriver()
 
-const mutationFollowUser = id => `
-  mutation {
-    follow(id: "${id}", type: User)
+let query
+let mutate
+let authenticatedUser
+
+let user1
+let user2
+let variables
+
+const mutationFollowUser = gql`
+  mutation($id: ID!, $type: FollowTypeEnum) {
+    follow(id: $id, type: $type) {
+      name
+      followedBy {
+        id
+        name
+      }
+      followedByCurrentUser
+    }
   }
 `
-const mutationUnfollowUser = id => `
-  mutation {
-    unfollow(id: "${id}", type: User)
+
+const mutationUnfollowUser = gql`
+  mutation($id: ID!, $type: FollowTypeEnum) {
+    unfollow(id: $id, type: $type) {
+      name
+      followedBy {
+        id
+        name
+      }
+      followedByCurrentUser
+    }
   }
 `
 
-beforeEach(async () => {
-  await factory.create('User', {
-    id: 'u1',
-    email: 'test@example.org',
-    password: '1234',
-  })
-  await factory.create('User', {
-    id: 'u2',
-    email: 'test2@example.org',
-    password: '1234',
+const userQuery = gql`
+  query($id: ID) {
+    User(id: $id) {
+      followedBy {
+        id
+      }
+      followedByCurrentUser
+    }
+  }
+`
+
+beforeAll(() => {
+  const { server } = createServer({
+    context: () => ({
+      driver,
+      user: authenticatedUser,
+      cypherParams: {
+        currentUserId: authenticatedUser ? authenticatedUser.id : null,
+      },
+    }),
   })
 
-  headersUser1 = await login({ email: 'test@example.org', password: '1234' })
-  clientUser1 = new GraphQLClient(host, { headers: headersUser1 })
+  const testClient = createTestClient(server)
+  query = testClient.query
+  mutate = testClient.mutate
+})
+
+beforeEach(async () => {
+  user1 = await factory
+    .create('User', {
+      id: 'u1',
+      name: 'user1',
+      email: 'test@example.org',
+      password: '1234',
+    })
+    .then(user => user.toJson())
+  user2 = await factory
+    .create('User', {
+      id: 'u2',
+      name: 'user2',
+      email: 'test2@example.org',
+      password: '1234',
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-16T19:42:43Z" title="Monday, September 16th 2019, 9:42:43 pm +02:00">Sep 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -41,44 +70,57 @@ describe('follow', () => {
   describe('follow user', () => {
     describe('unauthenticated follow', () => {
       it('throws authorization error', async () => {
-        const client = new GraphQLClient(host)
-        await expect(client.request(mutationFollowUser('u2'))).rejects.toThrow('Not Authorised')
+        authenticatedUser = null
+        const { errors } = await mutate({
+          mutation: mutationFollowUser('u2'),
+        })
+        expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
       })
     })
 
     it('I can follow another user', async () => {
-      const res = await clientUser1.request(mutationFollowUser('u2'))
-      const expected = {
-        follow: true,
-      }
-      expect(res).toMatchObject(expected)
+      const { data: result } = await mutate({
+        mutation: mutationFollowUser(user2.id),
+      })
+
+      const expected = { follow: true }
+      expect(result).toMatchObject(expected)
+
+      const {
+        data: { User },
+      } = await query({
```


Remove this whole workaround.